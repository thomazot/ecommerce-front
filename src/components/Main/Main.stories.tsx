import { Story, Meta } from '@storybook/react'
import Main from '.'

export default {
  title: 'Components/Main',
  component: Main,
  args: {
    children: 'Component Main'
  }
} as Meta

export const Default: Story = (args) => <Main {...args} />
