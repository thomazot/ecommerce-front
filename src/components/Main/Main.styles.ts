import styled from 'styled-components'
import tw from 'twin.macro'

export const Container = styled.main`
  ${tw`relative bg-white`}
`
