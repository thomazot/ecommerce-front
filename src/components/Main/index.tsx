const Main = () => (
  <main>
    <h1>react avanÃ§ados</h1>
  </main>
)

export default Main
