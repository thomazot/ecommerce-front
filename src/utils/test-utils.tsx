import { ReactElement } from 'react'
import { render, renderHook, RenderOptions } from '@testing-library/react'

import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'
import { StoreContextProvider } from 'stores'

type CustomRenderProps = Omit<RenderOptions, 'queries'>

const WrapperProvider: React.FC<{
  children?: React.ReactNode | React.ReactNode[]
}> = ({ children }) => (
  <StoreContextProvider>
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </StoreContextProvider>
)

const customRender = (
  ui: ReactElement,
  { ...renderOptions }: CustomRenderProps = {}
) => render(<WrapperProvider>ui</WrapperProvider>, renderOptions)

function customRenderHook<T>(useCustom: () => T) {
  return renderHook(() => useCustom(), { wrapper: WrapperProvider })
}

export * from '@testing-library/react'
export { customRender as render }
export { customRenderHook as renderHook }
