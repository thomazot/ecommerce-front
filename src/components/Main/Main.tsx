import * as S from './Main.styles'

export type MainProps = {
  children?: React.ReactNode | React.ReactNode[]
}

const Main: React.FC<MainProps> = ({ children }) => (
  <S.Container>
    <div className="mx-auto max-w-7xl px-4 sm:px-6 lg:px-8">{children}</div>
  </S.Container>
)

export default Main
