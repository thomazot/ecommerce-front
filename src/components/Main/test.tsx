import { screen, render } from '@testing-library/react'

import Main from './'

describe('<Main />', () => {
  it('should render the heading', () => {
    render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react avan√ßado/i })
    ).toBeInTheDocument()
  })
})
