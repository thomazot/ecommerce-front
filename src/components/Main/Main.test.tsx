import { render } from 'utils/test-utils'

import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
