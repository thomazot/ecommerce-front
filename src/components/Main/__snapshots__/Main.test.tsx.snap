// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  position: relative;
  --tw-bg-opacity: 1;
  background-color: rgba(255,255,255,var(--tw-bg-opacity));
}

<main
  class="c0"
>
  <div
    class="mx-auto max-w-7xl px-4 sm:px-6 lg:px-8"
  />
</main>
`
